crates/sui-types/src/temporary_store.rs

```diff
@@ -1494,7 +1512,11 @@ impl<S: ChildObjectResolver> Storage for TemporaryStore<S> {
 
 impl<S: BackingPackageStore> BackingPackageStore for TemporaryStore<S> {
     fn get_package_object(&self, package_id: &ObjectID) -> SuiResult<Option<Object>> {
-        self.store.get_package_object(package_id)
```

Comment:
I don't know if this change is going to have broader implications -- I think it might inadvertantly introduce incomplete support for calling into packages that you just published.  It would be better to limit this support to some kind of wrapper type that we only use for our purposes.

Reply:
Please double check where this is used to make sure the semantic change is ok

Reply:
@awelc thoughts about this change?